setup/setup-ci/security-scanner/README.md

```diff
@@ -0,0 +1,187 @@
+---
+page_type: sample
+languages:
+- bash
+- python
+products:
+- azure-machine-learning
+description: Sample setup script to scan Compute Instances for malware and security vulnerabilities
+---
+
+# Compute Instance Security Scanner
+
+[![license: MIT](https://img.shields.io/badge/License-MIT-purple.svg)](../../../LICENSE)
+
+A security scanner for Azure ML [Compute Instances](https://learn.microsoft.com/en-us/azure/machine-learning/concept-compute-instance) reporting malware and vulnerabilities in OS and Python packages to [Azure Log Analytics](https://learn.microsoft.com/en-us/azure/azure-monitor/logs/log-analytics-overview). For details on the vulnerability management process for the Azure Machine Learning service, see [Vulnerability Management](https://learn.microsoft.com/azure/machine-learning/concept-vulnerability-management).
+
+## Getting Started
+
+> Prerequisite: an Azure ML workspace with a Compute Instance running and [diagnostic logs](https://learn.microsoft.com/en-us/azure/machine-learning/monitor-azure-machine-learning) streaming to Log Analytics. See further down for alternative setups.
+
+- **Upload the scanner to Azure ML**: download [`amlsecscan.py`](amlsecscan.py), open [Azure ML Studio](https://ml.azure.com/), go to the Notebooks tab, and upload the file into your user folder `/Users/{user_name}` (replacing `{user_name}` with your user alias).
+- **Install the scanner**: open a terminal in Azure ML Notebooks and run `sudo ./amlsecscan.py install`
+- **Run a scan**: in the terminal, run `sudo ./amlsecscan.py scan all` (this takes a few minutes)
+
+## Telemetry
+
+In Log Analytics, the scanner reports hearbeats to table `AmlSecurityComputeHealth_CL` and assessment results to `AmlSecurityComputeAssessments_CL`.
+
+Examples of Log Analytics [KQL](https://docs.microsoft.com/en-us/azure/data-explorer/kql-quick-reference) queries:
+- Recent heartbeats and scan status: `AmlSecurityComputeHealth_CL | top 100 by TimeGenerated desc`
+- Recent assessments: `AmlSecurityComputeAssessments_CL | top 100 by TimeGenerated desc`
+
+Hearbeats are emitted every 10 minutes. Scans are scheduled either daily around 5AM UTC or 10 minutes after machine reboot.
+
+Assessments cover the following areas:
+Assessment|Source
+--|--
+Malware|[ClamAV](https://www.clamav.net/)
+OS + Python vulnerabilities|[Trivy](https://github.com/aquasecurity/trivy)
+
+## Installation
+
+> Note: on Compute Instances with `sudo` disabled (`rootAccess = false`), installation is only possible via [Setup Script](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-customize-compute-instance).
```

Comment:
Nice! Would love to know how to do this

Reply:
It's about using the feature for disabling root access. I cannot find any resources on how to do this.
I will take this comment offline, thanks!

Reply:
To get started, there is one example of `rootAccess` being set [here](https://github.com/Azure/azureml_run_specification/blob/471525064f7216fb4607e91c3fdba62be30b97e5/specs/ci-idle-shutdown.md). I will look for more docs and get back to you.